Deep cleaning datadelay

pratikunterwegs opened the pull request:
This PR aims to bring {datadelay} in line with general R package and Epiverse standards.

1. Fixes #30 by using CSL references instead of Bibtex,
2. Replaces the "Setup and installation" vignette with the standard "Get started vignette",
3. Uses {ggplot2} manual code to plot case, death, and known outcomes data in the vignettes; this allows the `plot_*()` functions to be safely removed,
4. Moves some data processing from `estimate_static()` to `estimate_severity()`,
5. Adds spell-checking,
6. Updates the Readme, removing bad plots, this is WIP #45,
7. Removes unnecessary package documentation, including importing the `.data` pronoun from {rlang},
8. Removes unused imports such as {epiparameter} and {zoo},
9. Swaps `stats::runmed()` in place of `zoo::rollmean()`, fixes #27,
10. Updates the title and description under DESCRIPTION,
11. Removes the functions `format_output()` (fixes #29, fixes #26) and `plot_epiparameter_distribution()` (fixes #28),
12. Removes the "location" argument from a number of functions,
13. Updates function documentation and arguments,
14. Adds input checking to top-level functions,
15. Adds tests for multiple functions, fixes #25,
16. Corrects documentation and tests for `known_outcomes()`.

Help wanted:
~~1. Could I please get feedback on tackling issue #42?~~ : See comments in issue

Bisaloo commented on R/estimate_reporting.R:
```diff
@@ -87,14 +119,16 @@ estimate_reporting <- function(df_in,
-    if (!is.null(max_date)) {
+    # collect the severity at the last date, or the date specified by
+    # the user in `max_date`
+    if (!missing(max_date)) {
```
Again, `max_date` can't be missing because no default is provided. The best option might be to restore the default

pratikunterwegs replied:
Fixed in https://github.com/epiverse-trace/datadelay/pull/44/commits/809c4557a1949bb4cc13f4516597076e0343f919 and others

Bisaloo commented on R/estimate_time_varying.R:
```diff
@@ -80,69 +80,74 @@ estimate_time_varying <- function(df_in,
+      "`epidist` object required when `correct_for_delays` is TRUE"
     )
   }
   if (!missing(epi_dist)) {
```
epi_dist can't be missing because no default is provided

pratikunterwegs replied:
Fixed in https://github.com/epiverse-trace/datadelay/pull/44/commits/809c4557a1949bb4cc13f4516597076e0343f919 and others

pratikunterwegs replied:
The function would still run if `correct_for_delays` is false though. I agree the intention is to correct for delays, so would you prefer the `epi_dist` argument to have no default then?

Bisaloo replied:
Yes, you're right, I missed that `epi_dist` is only used when `correct_for_delays` is true so it's probably fine to have a default there.

pratikunterwegs commented:
There appears to be an issue with {covidregionaldata} that is causing some workflows to fail - I will rerun those workflows soon and hope the issue is resolved, please go ahead and review the changes anyway.

jamesmbaazam commented:
@pratikunterwegs Thanks for tagging me. May I know by when you need this reviewed? I will not be able to complete it by close of this week.

joshwlambert commented:
Lots of really nice work in this PR @pratikunterwegs. I've reviewed and it looks good to me to be merged when you're ready. 

### Issues

- There is no contributing guide in the .github folder so the link in the README does not work.
- Remove explicit namespacing when using {scales} as other functions are not namespaced
- I don't see anywhere that {kableExtra} is being used so it can be removed from the vignette and suggests. Locally at least the vignette still renders without it. 

### General comments from reviewing:

- I like the improved package title and description
- The replacement of bibtex by CSL-JSON and removal of .csl files, and addition of spell checking is all good
- I couldn't find any issues with the modifications and additions to the testing, it is much improved from pre-PR and will require a future PR to increase coverage
- I'm not sure about: "_{{ packagename }}_ relies on the [_epiparameter_ package](https://epiverse-trace.github.io/epiparameter/)." I know it's been done with {finalsize}, but that related to non-R dependencies and linking. Not a problem to mention (or market) {epiparameter} as long as it is clear that the dependencies are automatically handled by the package and do not require the user to manually manage dependencies as it could put new R users off. This could be good point to discuss around how much we want to expose users to dependencies.
- The updates to the vignettes looks good, nice updates to use Epiverse-TRACE standards, good improvements in text.
- Could use  class notation `<>` for `epidist` in getting started vignette
- Inconsistent styling of package name, getting started uses italics, whereas estimate_ascertainment uses monospaced
- Updates to functions all look fine
- The `TODO` comment in  `estimate_severity()` about the magic number is the 95% CI see https://sites.warnercnr.colostate.edu/gwhite/profile-likelihood-confidence-intervals/#:~:text=For%20a%20single%20parameter%2C%20likelihood,with%201%20degree%20of%20freedom.
- I agree with the removal of `plot_epiparameter_distribution()` in favour of calling the {epiparameter} method 
- I agree with the removal of `format_output()`

pratikunterwegs commented:
Thanks for taking a look @joshwlambert.

1. Added a contributing guide,
2. Removed explicit namespacing with {scales},
3. {kableExtra} has been removed,
4. Edited the sentence on relying on {epiparameter} to be a bit more neutral - I think it's fair to expose users to this dependency as it's fairly key to using the correcting for delays functionality, and is also covered in the Readme example itself
5. Using `<>` to denote classes, and italics to denote package names,
6. Removed the TODO comment.

pratikunterwegs commented:
Merging this PR now, we would ideally schedule a full review for this package relatively soon.
